packages/aws-appsync/src/client.js

```diff
@@ -28,51 +84,47 @@ class AWSAppSyncClient extends ApolloClient {
 
     /**
      *
-     * @param {string} url
+     * @param {object} appSyncOptions
      * @param {ApolloClientOptions<InMemoryCache>} options
      */
-    constructor({ url, region, auth, conflictResolver, complexObjectsCredentials, disableOffline = false }, options) {
-        if (!url || !region || !auth) {
+    constructor({
+        url,
+        region,
+        auth,
+        conflictResolver,
+        complexObjectsCredentials,
+        cacheOptions,
+        disableOffline = false
+    } = {}, options = {}) {
+        const { cache: customCache, link: customLink } = options;
+
+        if (!customLink && (!url || !region || !auth)) {
             throw new Error(
                 'In order to initialize AWSAppSyncClient, you must specify url, region and auth properties on the config object.'
             );
         }
 
-        let res;
-        this.hydratedPromise = new Promise((resolve, reject) => {
-            res = resolve;
-        });
+        let resolveClient;
+
+        const store = disableOffline ? null : createStore(() => this, () => resolveClient(this), conflictResolver);
+        const cache = disableOffline ? (customCache || new InMemoryCache(cacheOptions)) : new OfflineCache(store, cacheOptions);
```

Comment:
What happen if someone use ```customCache``` with ```cacheOptions```

Reply:
In that case, `cacheOptions` would be ignored. It is inferred that your `customCache` is already initialized with whatever options you chose.